PurchasesCoreSwift/Public/PurchaserInfo.swift

```diff
@@ -0,0 +1,315 @@
+//
+//  PurchaserInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Madeline Beyl on 7/9/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCPurchaserInfo) public class PurchaserInfo: NSObject {
+
+    /// Entitlements attached to this purchaser info
+    @objc public let entitlements: EntitlementInfos
+
+    /// All *subscription* product identifiers with expiration dates in the future.
+    @objc public var activeSubscriptions: Set<String> { activeKeys(dates: expirationDatesByProductId) }
+
+    /// All product identifiers purchases by the user regardless of expiration.
+    @objc public lazy var allPurchasedProductIdentifiers: Set<String> = {
+        return Set(self.expirationDatesByProductId.keys).union(self.nonSubscriptionTransactions.map { $0.productId })
+    }()
+
+    /// Returns the latest expiration date of all products, nil if there are none
+    @objc public var latestExpirationDate: Date? {
+        let mostRecentDate = self.expirationDatesByProductId
+            .values
+            .compactMap { $0 }
+            .max { $0.timeIntervalSinceReferenceDate < $1.timeIntervalSinceReferenceDate }
+
+        return mostRecentDate
+    }
+
+    /// Returns all product IDs of the non-subscription purchases a user has made.
+    @available(*, deprecated, message: "use nonSubscriptionTransactions")
+    @objc public var nonConsumablePurchases: Set<String> { Set(self.nonSubscriptionTransactions.map { $0.productId }) }
+
+    /**
+    Returns all the non-subscription purchases a user has made.
+    The purchases are ordered by purchase date in ascending order.
+     */
+    @objc public let nonSubscriptionTransactions: [Transaction]
+
+    /**
+     Returns the fetch date of this Purchaser info.
+     Note: Can be `nil` if was cached before we added this
+     */
+    @objc public let requestDate: Date
+
+    /// The date this user was first seen in RevenueCat.
+    @objc public let firstSeen: Date
+
+    /// The original App User Id recorded for this user.
+    @objc public let originalAppUserId: String
+
+    /**
+     URL to manage the active subscription of the user.
+     If this user has an active iOS subscription, this will point to the App Store,
+     if the user has an active Play Store subscription it will point there.
+     If there are no active subscriptions it will be null.
+     If there are multiple for different platforms, it will point to the App Store
+     */
+    @objc public let managementURL: URL?
+
+    /**
+    Returns the purchase date for the version of the application when the user bought the app.
+    Use this for grandfathering users when migrating to subscriptions.
+
+    Note: This can be `nil`, see `RCPurchases restoreTransactionsWithCompletionBlock`
```

Comment:
https://github.com/realm/jazzy#supported-documentation-keywords

I think this would have to be `[RCPurchases restoreTransactionsWithCompletionBlock:]`, but then again, since we're migrating, perhaps we should use the swift syntax, i.e.: `Purchases.restoreTransactions(completionBlock:)`